Sources/DataPipeline/CIODataPipeline.swift

```diff
@@ -0,0 +1,173 @@
+import CioInternalCommon
+import Foundation
+import Segment
+
+public class CIODataPipeline {
+    // Private static instance of Analytics, created only once.
+    private static var analytics: Analytics = .init(configuration: Configuration.defaultConfiguration)
+
+    // Private initializer to enforce singleton usage, prevents external instantiation.
+    private init() {}
+
+    // Static method to access the singleton Analytics instance.
+    public static func shared() -> Analytics {
```

Comment:
I was expecting `CIODataPipeline.shared` to return an instance of `CIODataPipeline` but this is returning the Segment SDK instance. 

Do we need to publicly expose the Segment SDK? If so, is `shared` the right name for exposing it?

Reply:
if we are to use the instance of `Analytics` in extension method of `CustomerIO` we need to have it as a static member of `CIODataPipeline`. 

Open to naming it whatever makes the most sense, since this is not going to be accessed by the users directly, shared seemed like a good candidate.

Reply:
`shared` is a commonly used Swift name when naming singleton instances. I think that is what is throwing me off. If I saw Swift code `CIODataPipeline.shared`, I was assume that an instance of CIODataPipeline would be returned. 

Instead of having this `shared` function, could we instead re-use the static Segment variable above? 

```swift
public private(set) static var analytics: Analytics = .init(configuration: Configuration.defaultConfiguration)
```

Then, `CIODataPipeline.analytics` would do the same thing as `CIODataPipeline.shared` does currently.